src/main/java/apod/ApodPresenter.java

```diff
@@ -45,16 +48,16 @@ private void onNext(ApodData apodData) {
             } catch (IOException e) {
                 e.printStackTrace();
             }
-            view.setPhoto(image);
+            viewProvider.get().setPhoto(image);
```

Comment:
call viewProvider.get() only once and save it to a variable rather than calling it multiple times in this method.